src/subdomains/supporting/payin/strategies/register/impl/monero.strategy.ts

```diff
@@ -83,10 +85,19 @@ export class MoneroStrategy extends RegisterStrategy {
     return [...transferInResults].reverse().map((p) => ({
       address: BlockchainAddress.create(p.address, this.blockchain),
       txId: p.txid,
-      txType: null,
+      txType: this.getTxType(p),
       blockHeight: p.height,
       amount: p.amount,
       asset,
     }));
   }
+
+  private getTxType(transfer: MoneroTransferDto): PayInType | undefined {
+    if (transfer.destinations?.length === 1) {
+      const address = transfer.destinations[0].address;
+      return Util.equalsIgnoreCase(Config.payment.moneroAddress, address) ? PayInType.PAYMENT : PayInType.DEPOSIT;
+    }
+
+    return PayInType.DEPOSIT;
+  }

```

Comment:
There might theoretically be a payment with multiple outputs. Maybe you can just check every destination?
```ts
return transfer.destinations?.some((d) => Util.equalsIgnoreCase(Config.payment.moneroAddress, d.address)
   ? PayInType.PAYMENT
   : PayInType.DEPOSIT;
```